repose-aggregator/core/repose-core-api/src/main/java/org/openrepose/core/services/RequestProxyService.java

```diff
@@ -31,17 +31,38 @@
     //TODO: this is the most terrible return value ever
     int proxyRequest(String targetHost, HttpServletRequest request, HttpServletResponse response) throws IOException;
 
+    int proxyRequest(String targetHost, HttpServletRequest request, HttpServletResponse response, String connPoolId)
+            throws IOException;
+
     void setRewriteHostHeader(boolean value);
 
+    @Deprecated
```

Comment:
All of the deprecated methods have an equivalent method that just also happens to take a connection pool ID. The deprecated methods just call through to the equivalent method passing null as the connection pool ID. The result is an interface with quite a few methods. The alternative would be an interface with fewer methods, but which expected null arguments in certain cases. It should be noted that the connection pool service API expects null in certain cases.

Initially I had left a TODO, as I wasn't sure whether or not we would want to deprecate these. There was a comment chain that started earlier this week asking about the TODO, so I just went ahead and did the TODO. I can drop the deprecations if you prefer the API as-is.